src/fpmath.h

```diff
@@ -47,7 +47,7 @@
 #define _PDP_ENDIAN       __ORDER_PDP_ENDIAN__
 #define _BYTE_ORDER       __BYTE_ORDER__
 
-#elif defined(__linux)
+#if defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__GNU__)
```

Comment:
`__FreeBSD_kernel__` means that the system has a FreeBSD kernel, and we don't care about the userland, right? As in, it should be valid if even FreeBSD defined this, right? If so, then this change is wrong. FreeBSD does not have `<features.h>`, `<endian.h>`, etc.